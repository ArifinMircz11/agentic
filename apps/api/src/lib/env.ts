import 'dotenv/config'

import { z } from 'zod'

export const envSchema = z.object({
  NODE_ENV: z
    .enum(['development', 'test', 'production'])
    .default('development'),
  DATABASE_URL: z.string().url(),
  JWT_SECRET: z.string()
})

// eslint-disable-next-line no-process-env
export const env = envSchema.parse(process.env)

export const isProd = env.NODE_ENV === 'production'
