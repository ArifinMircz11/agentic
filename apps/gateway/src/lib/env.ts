import type { Simplify } from 'type-fest'
import { parseZodSchema } from '@agentic/platform-core'
import {
  envSchema as baseEnvSchema,
  parseEnv as parseBaseEnv
} from '@agentic/platform-hono'
import { z } from 'zod'

export const envSchema = baseEnvSchema
  .extend({
    AGENTIC_API_BASE_URL: z.string().url(),
    AGENTIC_API_KEY: z.string().nonempty(),

    DO_RATE_LIMITER: z.custom<DurableObjectNamespace>(
      (ns) => ns && typeof ns === 'object'
    )
  })
  .strip()
export type RawEnv = z.infer<typeof envSchema>
export type Env = Simplify<ReturnType<typeof parseEnv>>

export function parseEnv(inputEnv: Record<string, unknown>) {
  const baseEnv = parseBaseEnv({
    SERVICE: 'gateway',
    ...inputEnv
  })

  const env = parseZodSchema(
    envSchema,
    { ...inputEnv, ...baseEnv },
    {
      error: 'Invalid environment variables'
    }
  )

  return {
    ...baseEnv,
    ...env
  }
}
