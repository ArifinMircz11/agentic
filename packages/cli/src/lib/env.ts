export const agenticApiBaseUrl =
  process.env.AGENTIC_API_BASE_URL || 'https://api.agentic.so'
