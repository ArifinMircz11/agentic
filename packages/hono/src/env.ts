import 'dotenv/config'

import type { Simplify } from 'type-fest'
import { parseZodSchema } from '@agentic/platform-core'
import { z } from 'zod'

import { logLevelsSchema } from './logger'

export const envSchema = z
  .object({
    ENVIRONMENT: z
      .enum(['development', 'test', 'production'])
      .default('development'),

    SERVICE: z.enum(['api', 'gateway']),

    LOG_LEVEL: logLevelsSchema.default('info'),

    SENTRY_DSN: z.string().url().optional()
  })
  .strip()

export function parseEnv(inputEnv: unknown) {
  const env = parseZodSchema(envSchema, inputEnv, {
    error: 'Invalid environment variables'
  })

  const isDev = env.ENVIRONMENT === 'development'
  const isTest = env.ENVIRONMENT === 'test'
  const isProd = env.ENVIRONMENT === 'production'
  const isBrowser = (globalThis as any).window !== undefined

  if (isProd && !env.SENTRY_DSN) {
    throw new Error(
      'Internal error: missing required "SENTRY_DSN" environment variable in production'
    )
  }

  return {
    ...env,
    isDev,
    isTest,
    isProd,
    isBrowser
  }
}

export type Env = Simplify<ReturnType<typeof parseEnv>>
