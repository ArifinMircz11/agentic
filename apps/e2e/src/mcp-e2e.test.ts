import { Client as McpClient } from '@modelcontextprotocol/sdk/client/index.js'
import { StreamableHTTPClientTransport } from '@modelcontextprotocol/sdk/client/streamableHttp.js'
import { afterAll, beforeAll, describe, expect, test } from 'vitest'

import { env } from './env'
import { fixtureSuites } from './mcp-fixtures'

for (const [i, fixtureSuite] of fixtureSuites.entries()) {
  const { title, fixtures, compareResponseBodies = false } = fixtureSuite

  const describeFn = fixtureSuite.only ? describe.only : describe
  describeFn(title, () => {
    let fixtureResult: any | undefined

    let client: McpClient
    beforeAll(async () => {
      client = new McpClient({
        name: fixtureSuite.path,
        version: '0.0.0'
      })

      const transport = new StreamableHTTPClientTransport(
        new URL(fixtureSuite.path, env.AGENTIC_GATEWAY_BASE_URL)
      )
      await client.connect(transport)
    }, 120_000)

    afterAll(async () => {
      await client.close()
    })

    for (const [j, fixture] of fixtures.entries()) {
      const {
        isError,
        result: expectedResult,
        content: expectedContent,
        structuredContent: expectedStructuredContent,
        _meta: expectedMeta,
        validate
      } = fixture.response ?? {}
      const snapshot =
        fixture.response?.snapshot ?? fixtureSuite.snapshot ?? !isError
      const debugFixture = !!(
        fixture.debug ??
        fixtureSuite.debug ??
        fixture.only ??
        fixtureSuite.only
      )
      const fixtureName = `${i}.${j}: ${fixtureSuite.path} ${fixture.request.name}`

      let testFn = (fixture.only ?? fixture.debug) ? test.only : test
      if (fixtureSuite.sequential) {
        testFn = testFn.sequential
      }

      testFn(
        fixtureName,
        {
          timeout: fixture.timeout ?? 60_000
        },
        // eslint-disable-next-line no-loop-func
        async () => {
          const { tools } = await client.listTools()
          // console.log('tools', tools)
          expect(tools.map((t) => t.name)).toContain(fixture.request.name)

          const result = await client.callTool({
            name: fixture.request.name,
            arguments: fixture.request.args
          })

          if (debugFixture) {
            console.log(fixtureName, '=>', result)
          }

          if (isError) {
            expect(result.isError).toBeTruthy()
          } else {
            expect(result.isError).toBeFalsy()
          }

          if (expectedResult) {
            expect(result).toEqual(expectedResult)
          }

          if (expectedContent) {
            expect(result.content).toEqual(expectedContent)
          }

          if (expectedStructuredContent) {
            expect(result.structuredContent).toEqual(expectedStructuredContent)
          }

          if (expectedMeta) {
            expect(result._meta).toBeDefined()
            for (const [key, value] of Object.entries(expectedMeta)) {
              expect(result._meta![key]).toEqual(value)
            }
          }

          if (snapshot) {
            expect(result).toMatchSnapshot()
          }

          if (validate) {
            await Promise.resolve(validate(result))
          }

          if (compareResponseBodies && !isError) {
            if (!fixtureResult) {
              fixtureResult = result
            } else {
              expect(result).toEqual(fixtureResult)
            }
          }
        }
      )
    }
  })
}
