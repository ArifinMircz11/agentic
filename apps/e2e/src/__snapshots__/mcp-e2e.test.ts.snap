// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Basic MCP => OpenAPI @ 010332cf get_post success  > 2.0: @dev/test-basic-openapi@010332cf/mcp get_post 1`] = `
{
  "content": [],
  "isError": false,
  "structuredContent": {
    "body": "dignissimos aperiam dolorem qui eum
facilis quibusdam animi sint suscipit qui sint possimus cum
quaerat magni maiores excepturi
ipsam ut commodi dolor voluptatum modi aut vitae",
    "id": 8,
    "title": "dolorem dolore est ipsam",
    "userId": 1,
  },
}
`;

exports[`Basic MCP => OpenAPI @ latest get_post success  > 1.0: @dev/test-basic-openapi@latest/mcp get_post 1`] = `
{
  "content": [],
  "isError": false,
  "structuredContent": {
    "body": "et iusto sed quo iure
voluptatem occaecati omnis eligendi aut ad
voluptatem doloribus vel accusantium quis pariatur
molestiae porro eius odio et labore et velit aut",
    "id": 3,
    "title": "ea molestias quasi exercitationem repellat qui ipsa sit aut",
    "userId": 1,
  },
}
`;

exports[`Basic MCP => OpenAPI get_post success > 0.0: @dev/test-basic-openapi/mcp get_post 1`] = `
{
  "content": [],
  "isError": false,
  "structuredContent": {
    "body": "quia et suscipit
suscipit recusandae consequuntur expedita et cum
reprehenderit molestiae ut ut quas totam
nostrum rerum est autem sunt rem eveniet architecto",
    "id": 1,
    "title": "sunt aut facere repellat provident occaecati excepturi optio reprehenderit",
    "userId": 1,
  },
}
`;
